src/Integration.UnitTests/CommandInterceptorTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Design;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests
+{
+    [TestClass]
+    public class CommandInterceptorTests
+    {
+        [TestMethod]
+        public void Exec_CommandProgressionStops_ReturnsZero()
+        {
+            var guid = new Guid();
+            var commandID = new CommandID(guid, 0);
+
+            var func = () => { return CommandProgression.Stop; };
```

Comment:
> Here it would probably make sense to check if the functions are called, I am just not quite sure how without making it ugly.

One simple way is to create a local variable and set it in the test:

```
bool funcCalled = false;
var func = () =>
  {
    funcCalled = true;
    return CommandProgression.Stop;
  }

...


funcCalled.Should().BeTrue();
```